cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/services/AppFabricServer.java

```diff
@@ -131,14 +131,16 @@ protected void startUp() throws Exception {
         provisioningService.start(),
         applicationLifecycleService.start(),
         bootstrapService.start(),
-        systemAppManagementService.start(),
         programRuntimeService.start(),
         programNotificationSubscriberService.start(),
         runRecordCorrectorService.start(),
         coreSchedulerService.start()
       )
     ).get();
 
+    // Start system app management service only after bootstrap service is started.
+    systemAppManagementService.start().get();
```

Comment:
it's generally not a good idea to force a certain start order for services. It makes it very easy to break things by coupling together what would otherwise be unrelated services.

I think it would be better if we move loading of system artifacts from the bootstrap service to the system management service. I believe it makes more sense there anyway. If an admin wants to start a new system app, they may need to reload system artifacts anyway.